forte/proc/dsrg.py

```diff
@@ -254,26 +256,34 @@ def compute_energy(self):
 
             # Continue to solve DSRG equations
 
-            # - Compute RDMs (RDMs available if done relaxed dipole)
+            # - Compute RDMs from the active space solver (the RDMs are already available if we computed the relaxed dipole)
+            #   These RDMs are computed in the original basis
             if self.do_multi_state or (not self.do_dipole):
                 self.rdms = self.active_space_solver.compute_average_rdms(self.state_weights_map, self.max_rdm_level,
                                                                           self.rdm_type)
 
-            # - Transform RDMs to the semi-canonical orbitals of last step (because of integrals)
+            # - Transform RDMs to the semi-canonical basis used in the last step (stored in self.Ua/self.Ub)
+            #   We do this because the integrals and amplitudes are all expressed in the previous semi-canonical basis
             self.rdms.rotate(self.Ua, self.Ub)
 
             # - Semi-canonicalize RDMs and orbitals
             if self.do_semicanonical:
                 self.semi.semicanonicalize(self.rdms)
-                # NOT read previous orbitals if fixing orbital ordering and phases failed
+                # Do NOT read previous orbitals if fixing orbital ordering and phases failed
                 if (not self.semi.fix_orbital_success()) and self.Heff_implemented:
                     psi4.core.print_out("\n  DSRG checkpoint files removed due to the unsuccessful"
                                         " attempt to fix orbital phase and order.")
                     self.dsrg_solver.clean_checkpoints()
-            self.Ua["ik"] = self.Ua["ij"] * self.semi.Ua_t()["jk"]
-            self.Ub["ik"] = self.Ub["ij"] * self.semi.Ub_t()["jk"]
 
-            # - Compute DSRG energy
+                # update the orbital transformation matrix that connects the original orbitals
```

Comment:
Question for @lcyyork: indenting the lines below seems to be the right thing to do (apply Ua from semi to the current set of Ua only if we do indeed semicanonicalize). Otherwise we would be always rotating by some quantity (I assume the identity, but it's not good to introduce some indirect dependency on the internals of the semi class).

Reply:
Yes, it is identity matrix by default.

Reply:
> Question for @lcyyork: indenting the lines below seems to be the right thing to do (apply Ua from semi to the current set of Ua only if we do indeed semicanonicalize). Otherwise we would be always rotating by some quantity (I assume the identity, but it's not good to introduce some indirect dependency on the internals of the semi class).

Yes, it is identity matrix by default. The semi class is initialized no matter actual semi computations or not.